cmd/erasure-object.go

```diff
@@ -1497,12 +1525,18 @@ func (er erasureObjects) DeleteObject(ctx context.Context, bucket, object string
 			// any version specified explicitly. Or if a particular
 			// version id needs to be replicated.
 			if err = er.deleteObjectVersion(ctx, bucket, object, writeQuorum, fi, opts.DeleteMarker); err != nil {
-				return objInfo, toObjectErr(err, bucket, object)
+				// soft deletes are idempotent. Return the topmost delete marker in stack if delete marker already set
+				if !errors.As(err, &errDeleteNoOp) {
+					return objInfo, toObjectErr(err, bucket, object)
+				}
+				oi, err := er.getObjectInfo(ctx, bucket, object, ObjectOptions{})
```

Comment:
why do we still need to do this? @poornas can we not get FileInfo back again after deleteObjectVersion() ?